libraries/botframework-connector/src/auth/credentialProvider.ts

```diff
@@ -58,6 +58,11 @@ export class SimpleCredentialProvider implements ICredentialProvider {
     private readonly appId: string;
     private readonly appPassword: string;
 
+    // Protects against JSON.stringify leaking secrets
+    private toJSON(): unknown {
+        return { name: this.constructor.name, appId: this.appId };
```

Comment:
Did we want to avoid `this.constructor.name`? I know we did in other areas in the event of minification; not sure if that applies here.

Reply:
I think using it as a key in turn state is problematic because users may interchangeably use `this.constructor.name` and the actual string key. If there is some minification that happens, the string keys will no longer work. In this case, I'm not worried about `JSON.stringify` output.